pkg/kubelet/dockertools/manager.go

```diff
@@ -573,13 +573,21 @@ func makeEnvList(envs []kubecontainer.EnvVar) (result []string) {
 // can be understood by docker.
 // Each element in the string is in the form of:
 // '<HostPath>:<ContainerPath>', or
-// '<HostPath>:<ContainerPath>:ro', if the path is read only.
-func makeMountBindings(mounts []kubecontainer.Mount) (result []string) {
+// '<HostPath>:<ContainerPath>:ro', if the path is read only, and/or
+// '<HostPath>:<ContainerPath>:Z', if the volume requires SELinux
+// relabling and the pod provides an SELinux label
+func makeMountBindings(mounts []kubecontainer.Mount, podHasSELinuxLabel bool) (result []string) {
 	for _, m := range mounts {
 		bind := fmt.Sprintf("%s:%s", m.HostPath, m.ContainerPath)
 		if m.ReadOnly {
 			bind += ":ro"
 		}
+		// If the pod does not provide an SELinux context
+		// relabeling will restrict access to the volume the
+		// container which mounts it last
+		if m.SELinuxRelabel && podHasSELinuxLabel {
```

Comment:
We only want to do this for the first container that mounts the volume, so we need to have the pod and the containers passed into this method.  We should only relabel each volume a single time.